### PR TITLE
feat: report snap version without vc

### PR DESCRIPTION
**Pull request by dawnseeker8**
## Description
This PR will implement the UI component for `report snap version` story.
Following tasks have been completed:
1. Create component `SnapReport` to centrally control `ReportButton` and `ReportSnapModal`
2. Create component `ReportSnapModal` to user signing interaction
3. Modify `{Snap.slug}.test.tsx` to include condition check to show or hide the `SnapReport` component.

Screenshot 2024-01-29 at 15 53 13

Screenshot 2024-01-29 at 15 54 59

Screenshot 2024-01-29 at 15 54 02

### Related ticket

Fixes #
issue 45

### Type of change

- [ ] Chore
- [ ] Bug fix
- [x] New feature
- [ ] Documentation update

### CODE REVIEW

File: src/components/ReportButton.tsx [hunk 1]
```diff
@@ -7,17 +7,19 @@ import { DangerIcon } from '.';
 export type ReportButtonProps = {
   onClick: () => void;
   reported: boolean;
+  size?: 'sm' | 'md' | 'lg';
```
Comment (dawnseeker8) [inline src/components/ReportButton.tsx:10]: added to support make customise the Report button size, current version is very small to show in snap slug screen

File: src/pages/snap/{Snap.location}/{Snap.slug}.tsx [hunk 2]
```diff
@@ -1,18 +1,20 @@
 import { Box, Container, Divider, Flex, Text } from '@chakra-ui/react';
 import { Trans } from '@lingui/macro';
 import { graphql } from 'gatsby';
-import type { FunctionComponent } from 'react';
+import { type FunctionComponent } from 'react';
```
Comment (dawnseeker8) [inline src/pages/snap/{Snap.location}/{Snap.slug}.tsx:4]: This line is auto formatted by prettier based on rule.

File: src/pages/snap/{Snap.location}/{Snap.slug}.test.tsx [hunk 3]
```diff
@@ -8,16 +9,47 @@ import {
...
 describe('Snap page', () => {
+  let mockUseAccount: jest.Mock;
+
+  beforeEach(() => {
+    mockUseAccount = useAccount as jest.Mock;
+    mockUseAccount.mockClear();
+  });
+
   it('renders', async () => {
+    mockUseAccount.mockReturnValue({ isConnected: true });
```
Comment (stanleyyconsensys) [inline src/pages/snap/{Snap.location}/{Snap.slug}.test.tsx:25]: nic:

add space after mock, apply to all below

File: src/features/snap/components/SnapReport.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import { useState, type FunctionComponent } from 'react';
+
+import { ReportSnapModal } from './modals/ReportSnapModal';
+import { ReportButton } from '../../../components';
+
+type SnapReportProps = {
+  snapName: string;
```
Comment (stanleyyconsensys) [inline src/features/snap/components/SnapReport.tsx:7]: you need the snap id too, but i think you can add it in other PR

File: src/features/snap/components/SnapReport.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
+import { useState, type FunctionComponent } from 'react';
```
Comment (stanleyyconsensys) [inline src/features/snap/components/SnapReport.tsx:1]: the file naming should be ReportSnap

**Review by stanleyyconsensys — CHANGES_REQUESTED** [review]
LGTM, some comments

File: src/features/snap/components/SnapReport.tsx (hunk 4 above)
Comment (dawnseeker8) [reply src/features/snap/components/SnapReport.tsx:7]: lets do it another PR, i will do that when VC needed.

File: src/features/snap/components/SnapReport.tsx (hunk 5 above)
Comment (dawnseeker8) [reply src/features/snap/components/SnapReport.tsx:1]: I saw @satyajeetkolhapure file called `AccountReport`, that is why i create a similar name called `SnapReport` here.

File: src/pages/snap/{Snap.location}/{Snap.slug}.test.tsx (hunk 3 above)
Comment (dawnseeker8) [reply src/pages/snap/{Snap.location}/{Snap.slug}.test.tsx:25]: have changed all area with mockUserAccount with space between mockUseAccount and actual test.

File: src/features/snap/components/SnapReport.tsx (hunk 5 above)
Comment (dawnseeker8) [reply src/features/snap/components/SnapReport.tsx]: Just change file name to `ReportSnap`

**Review by ritave — APPROVED** [review]

**Pull request merged**